server/sandbox_run_linux.go

```diff
@@ -1026,21 +1045,21 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 	}
 
 	if isContextError(ctx.Err()) {
-		if err := s.resourceStore.Put(sbox.Name(), sb, resourceCleaner); err != nil {
-			log.Errorf(ctx, "RunSandbox: failed to save progress of sandbox %s: %v", sbox.ID(), err)
+		if err := s.resourceStore.Put(sandboxName, sb, resourceCleaner); err != nil {
+			log.Errorf(ctx, "RunSandbox: failed to save progress of sandbox %s: %v", sandboxID, err)
 		}
 		log.Infof(ctx, "RunSandbox: context was either canceled or the deadline was exceeded: %v", ctx.Err())
 		return nil, ctx.Err()
 	}
 
 	// Since it's not a context error, we can delete the resource from the store, it will be tracked in the server from now on.
-	s.resourceStore.Delete(sbox.Name())
+	s.resourceStore.Delete(sandboxName)
 
 	sb.SetCreated()
 	s.generateCRIEvent(ctx, sb.InfraContainer(), types.ContainerEventType_CONTAINER_STARTED_EVENT)
 
 	log.Infof(ctx, "Ran pod sandbox %s with infra container: %s", container.ID(), container.Description())
-	resp = &types.RunPodSandboxResponse{PodSandboxId: sbox.ID()}
+	resp = &types.RunPodSandboxResponse{PodSandboxId: sandboxID}
```

Comment:
I think for this PR I'd prefer to keep this sbox.ID() -> sandboxID (and Name similarly) out of it so we can focus on the structural changes if you wouldn't mind

Reply:
We delete the sandbox hence if we use sbox.ID() it tends to cause problems. Same for sbox.Name()